apps/src/code-studio/pd/application/principalApproval1819/PrincipalApproval1819Application.jsx

```diff
@@ -0,0 +1,235 @@
+      ...super.getPageProps(),
+      teacherApplication: this.props.teacherApplication
+    };
+  }
+
+  /**
+   * @override
+   */
+  serializeFormData() {
+    return {
+      ...super.serializeFormData(),
+      application_guid: this.props.teacherApplication.application_guid
+    };
+  }
+
+  onSuccessfulSubmit() {
+    window.location.reload(true);
+  }
+}
+
+class PrincipalApprovalComponent extends ApplicationFormComponent {
+  static labels = PageLabels;
+
+  static associatedFields = [
+    ...Object.keys(PageLabels)
+  ];
+
+  handleSchoolChange = selectedSchool => {
+    this.handleChange({school: selectedSchool && selectedSchool.value});
+  };
+
+  renderSchoolSection() {
+    // TODO: Mehal - this should be a seperate component
+    return (
+      <div>
+        <FormGroup
+          id="school"
+          controlId="school"
+          validationState={this.getValidationState("school")}
+        >
+          <Row>
+            <Col md={6}>
+              <ControlLabel>
+                School
+                <span style={{color: 'red'}}> *</span>
+              </ControlLabel>
+            </Col>
+          </Row>
+          <Row>
+            <Col md={6}>
+              <SchoolAutocompleteDropdown
+                value={this.props.data.school}
+                onChange={this.handleSchoolChange}
+              />
+            </Col>
+          </Row>
+        </FormGroup>
+        {
+          this.props.data.school && this.props.data.school === '-1' &&
+          <div>
+            {this.inputFor("schoolName")}
+            {this.inputFor("schoolAddress")}
+            {this.inputFor("schoolCity")}
+            {this.selectFor("schoolState", {placeholder: "Select a state"})}
+            {this.inputFor("schoolZipCode")}
+            {this.radioButtonsFor("schoolType")}
+          </div>
+        }
+      </div>
+    );
+  }
+
+  renderCourseReplacementSection() {
+    if (this.props.teacherApplication.course === 'Computer Science Discoveries') {
+      return this.checkBoxesWithAdditionalTextFieldsFor('replaceWhichCourseCsp', {
+        "Other (Please Explain):" : "other"
+      });
+    } else if (this.props.teacherApplication.course === 'Computer Science Principles') {
+      return this.checkBoxesWithAdditionalTextFieldsFor('replaceWhichCourseCsp',{
+        "Other (Please Explain):" : "other"
+      });
+    }
+  }
+
+  render() {
+    return (
+      <FormGroup>
+        <p>
+          Thank you for your support of computer science education! A teacher at your
+          school, {this.props.teacherApplication.name} has applied to be a part of
+          Code.org’s Professional Learning Program in order to teach the {' '}
+          {this.props.teacherApplication.course} curriculum during the 2018-19  school
+          year. Your completion of this survey is required for the teacher’s application
+          to be considered.
+        </p>
+        {this.inputFor('firstName')}
+        {this.inputFor('lastName')}
```

Comment:
The teacher supplied principal first and last name. Should we display those as initial values?